shared-libs/transitions/src/transitions/create_user_for_contacts.js

```diff
@@ -0,0 +1,127 @@
+const config = require('../config');
+const db = require('../db');
+const contactTypeUtils = require('@medic/contact-types-utils');
+const { people } = require('@medic/contacts')(config, db);
+const { users } = require('@medic/user-management')(config, db);
+
+const NAME = 'create_user_for_contacts';
+
+const UserCreationStatus = {
+  // PENDING - Set by webapp while waiting for sync to complete
+  READY: 'READY', // Ready to be replaced
+  COMPLETE: 'COMPLETE', // The new user has been created
+  ERROR: 'ERROR', // The new user could not be created
+};
+
+const getNewContact = (contactId) => {
+  if (!contactId) {
+    return Promise.reject(new Error('No id was provided for the new replacement contact.'));
+  }
+  return people.getOrCreatePerson(contactId);
+};
+
+const generateUsername = (contactName) => {
+  const randomNum = Math.floor(Math.random() * 9999).toString().padStart(4, '0');
+  const username = contactName.normalize('NFD') // split an accented letter in the base letter and the accent
+    .replace(/[\u0300-\u036f]/g, '') // remove all previously split accents
+    .toLowerCase()
+    .trim()
+    .replace(/[^a-z0-9 ]/g, '') // remove all chars not letters, numbers and spaces (to be replaced)
+    .replace(/\s+/g, '-'); // separator
+
+  return `${username}-${randomNum}`;
+};
+
+const generateUniqueUsername = (contactName) => {
+  const username = generateUsername(contactName);
+  return db.users.get(`org.couchdb.user:${username}`)
+    .then(() => generateUniqueUsername(contactName))
```

Comment:
Updated this logic so that after every 10 collisions it will add another digit to the suffix.  I have also added a max limit of 100 retries so that we will not keep trying forever...